azure-cli/monitor/alert.utilities.ps1

```diff
@@ -0,0 +1,259 @@
+. "$PSScriptRoot\..\utilities\deploy.utilities.ps1"
+
+function New-Topic-Condition
+{
+    param (
+        [Parameter(Mandatory = $true)]
+        [ValidateNotNullOrEmpty()]
+        [string]
+        $dimension,
+
+        [Parameter(Mandatory = $true)]
+        [ValidateNotNullOrEmpty()]
+        [string]
+        $threshold
+    )
+
+    return az monitor metrics alert condition create `
+    -t static `
+    --dimension $dimension `
+    --aggregation Average `
+    --metric "ActiveMessages" `
+    --op GreaterThan `
+    --threshold $threshold
+}
+
+function New-Dimension
+{
+    param (
+        [Parameter(Mandatory = $true)]
+        [ValidateNotNullOrEmpty()]
+        [string]
+        $entity
+    )
+
+    return az monitor metrics alert dimension create `
+    -n "EntityName" `
+    -v $entity
+}
+
+function Get-ResourceId {
+    param (
+        [Parameter(Mandatory=$true)]
+        [ValidateNotNullOrEmpty()]
+        [string]
+        $resourceGroupName,
+    
+        [Parameter(Mandatory=$true)]
+        [ValidateNotNullOrEmpty()]
+        [string] 
+        $serviceName,
+
+        [Parameter(Mandatory=$true)]
+        [ValidateNotNullOrEmpty()]
+        [string] 
+        $resourceType
+    )
+    $resourceId = az resource show `
+        --resource-group $resourceGroupName `
+        --name $serviceName `
+        --resource-type $resourceType `
+        --query id
+    return $resourceId.Replace('"','')
+}
+
+function New-ActionGroup{
+    param (
+        [Parameter(Mandatory=$true)]
+        [ValidateNotNullOrEmpty()]
+        [string]
+        $resourceGroupName,
+
+        [Parameter(Mandatory=$true)]
+        [ValidateNotNullOrEmpty()]
+        [string] 
+        $actionGroupName,
+
+        [Parameter(Mandatory=$true)]
+        [ValidateNotNullOrEmpty()]
+        [string] 
+        $actionGroupShortName,
+
+        [Parameter(Mandatory=$false)]
+        [string[]] 
+        $tags = @()
+    )
+
+    return az monitor action-group create `
+        --resource-group $resourceGroupName `
+        --name $actionGroupName `
+        --short-name $actionGroupShortName `
+        --tags $tags
+}
+
+function Add-EmailAction{
```

Comment:
Maybe we should add a similar function for adding a webhook - something like.

function Add-EmailAction{
    param (
        [Parameter(Mandatory=$true)]
        [ValidateNotNullOrEmpty()]
        [string]
        $resourceGroupName,

        [Parameter(Mandatory=$true)]
        [ValidateNotNullOrEmpty()]
        [string] 
        $actionGroupName,

        [Parameter(Mandatory=$true)]
        [ValidateNotNullOrEmpty()]
        [string] 
        $emailName,

        [Parameter(Mandatory=$true)]
        [ValidateNotNullOrEmpty()]
        [string] 
        $alertUri
    )

    return az monitor action-group update `
    --resource-group $resourceGroupName `
    --name $actionGroupName `
    --add-action webhook alert_hook $alertUri usecommonalertschema
}

Reply:
Added.